examples/knative/powershell/kn-ps-vrni-databus/handler.ps1

```diff
@@ -0,0 +1,84 @@
+Import-Module CloudEvents.Sdk
+
+. ./vrni-functions.ps1
+
+Function Process-Init {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Init`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   [CmdletBinding()]
+   param(
+      [Parameter(Position = 0, Mandatory = $true)]$Source,
+      [Parameter(Position = 1, Mandatory = $true)]$Body
+   )
+
+   # Retrieve webhook sink URL
+   $webhookSinkUrl = ${env:WEBHOOK_SINK_URL}
+
+   Write-host "$(Get-Date) - Creating CloudEvents...`n"
+   try {
+      $cloudEvents = New-vRNICloudEventsFromDatabus -Data $body -cloudEventSource $Source
+   }
+   catch {
```

Comment:
Since this does not throw and continue execution, is the following check on the count sufficient to terminate the function in case of invalid CloudEvent? Or should you rather throw immediately in this catch?

Reply:
Both, I think. The count is still there to prevent processing an empty payload. I've updated it to throw in the catch to exit out the script if there are errors in the payload and left the count for when the payload is valid but empty.